src/features/cookies/pages/CookiesConsent.native.test.tsx

```diff
@@ -22,75 +26,83 @@ describe('<CookiesConsent/>', () => {
   beforeEach(() => storage.clear(COOKIES_CONSENT_KEY))
 
   it('should render correctly', async () => {
-    const renderAPI = render(<CookiesConsent visible={true} hideModal={hideModal} />)
+    const renderAPI = renderCookiesConsent()
     expect(renderAPI).toMatchSnapshot()
   })
 
   describe('accept all cookies', () => {
-    it('should save cookies consent information in storage when accepted all cookies', async () => {
-      const { getByText } = render(<CookiesConsent visible={true} hideModal={hideModal} />)
+    it('should save cookies consent information in storage and log choice when accepted all cookies', async () => {
+      const { getByText } = renderCookiesConsent()
       const acceptAllButton = getByText('Tout accepter')
 
       fireEvent.press(acceptAllButton)
 
       await flushAllPromisesWithAct()
 
-      expect(await storage.readObject(COOKIES_CONSENT_KEY)).toEqual({
+      const storageContent = {
         deviceId,
         choiceDatetime: Today.toISOString(),
         consent: {
           mandatory: COOKIES_BY_CATEGORY.essential,
           accepted: ALL_OPTIONAL_COOKIES,
           refused: [],
         },
-      })
+      }
+      expect(await storage.readObject(COOKIES_CONSENT_KEY)).toEqual(storageContent)
+      await waitForExpect(() =>
+        expect(api.postnativev1cookiesConsent).toBeCalledWith(storageContent)
+      )
     })
 
     it('should hide modale when accepted all cookies', async () => {
-      const { getByText } = render(<CookiesConsent visible={true} hideModal={hideModal} />)
+      const { getByText } = renderCookiesConsent()
       const acceptAllButton = getByText('Tout accepter')
 
       fireEvent.press(acceptAllButton)
+      await flushAllPromisesWithAct()
 
       expect(hideModal).toBeCalled()
     })
   })
 
   describe('refuse all cookies', () => {
-    it('should save cookies consent information in storage when user refused all cookies', async () => {
-      const { getByText } = render(<CookiesConsent visible={true} hideModal={hideModal} />)
+    it('should save cookies consent information in storage and log choice when user refused all cookies', async () => {
```

Comment:
`and` dans la règle d'un test est un codesmell qui me dit qu'on vérifie 2 choses à la fois

je déconseille de faire des tests qui vérifient plusieurs choses à la fois

je recommande de découper en plusieurs tests unitaires